drivers/at86rf215/at86rf215_netdev.c

```diff
@@ -0,0 +1,922 @@
+    if (dev->flags & AT86RF215_OPT_TELL_RX_START) {
+        bb_irqs_enabled |= BB_IRQ_RXAM;
+    }
+
+    rf_irq_mask = at86rf215_reg_read(dev, dev->RF->RG_IRQS);
+    bb_irq_mask = at86rf215_reg_read(dev, dev->BBC->RG_IRQS);
+
+    uint8_t timeout = dev->timeout;
+    if (timeout) {
+        dev->timeout = 0;
+    }
+
+    /* mark AGC Hold bit */
+    if (bb_irq_mask & BB_IRQ_AGCH) {
+        dev->flags |= AT86RF215_OPT_AGCH;
+    }
+
+    /* clear AGC Hold bit */
+    if (bb_irq_mask & BB_IRQ_AGCR) {
+        dev->flags &= ~AT86RF215_OPT_AGCH;
+    }
+
+    /* we got here because of CMSA timeout */
+    if (timeout & AT86RF215_TIMEOUT_CSMA) {
+        timeout = 0;
+
+        if (!(dev->flags & AT86RF215_OPT_CCATX)) {
+            at86rf215_reg_write(dev, dev->RF->RG_EDC, 1);
+        } else {
+            at86rf215_rf_cmd(dev, CMD_RF_TXPREP);
+        }
+    }
+
+    /* If the interrupt pin is still high, there was an IRQ on the other radio */
+    if (gpio_read(dev->params.int_pin)) {
+        if (dev->sibling && dev->sibling->state != AT86RF215_STATE_OFF) {
+            netdev->event_callback((netdev_t *) dev->sibling, NETDEV_EVENT_ISR);
+        } else {
+            _clear_sibling_irq(dev);
+        }
+    }
+
+    /* exit early if the interrupt was not for this interface */
+    if (!((bb_irq_mask & bb_irqs_enabled) ||
+          (rf_irq_mask & (RF_IRQ_EDC | RF_IRQ_TRXRDY)) || timeout)) {
+        return;
+    }
+
+    /* check if the received packet has the ACK request bit set */
+    bool rx_ack_req;
+    if (bb_irq_mask & BB_IRQ_RXFE) {
+        rx_ack_req = at86rf215_reg_read(dev, dev->BBC->RG_FBRXS) & IEEE802154_FCF_ACK_REQ;
+    } else {
+        rx_ack_req = 0;
+    }
+
+    if (dev->flags & AT86RF215_OPT_CCA_PENDING) {
+
+        /* Start ED or handle result */
+        if (rf_irq_mask & RF_IRQ_EDC) {
+            _handle_edc(dev);
+        } else if (rf_irq_mask & RF_IRQ_TRXRDY) {
+            /* disable baseband for energy detection */
+            at86rf215_disable_baseband(dev);
+            /* switch to state RX for energy detection */
+            at86rf215_rf_cmd(dev, CMD_RF_RX);
+            /* start energy measurement */
+            at86rf215_reg_write(dev, dev->RF->RG_EDC, 1);
+        }
+
+    } else if (dev->flags & AT86RF215_OPT_TX_PENDING) {
+
+        /* start transmitting the frame */
+        if (rf_irq_mask & RF_IRQ_TRXRDY) {
+
+            /* automatically switch to RX when TX is done */
+            _enable_tx2rx(dev);
+
+            /* only listen for ACK frames */
+            if (dev->flags & AT86RF215_OPT_ACK_REQUESTED) {
+                at86rf215_filter_ack(dev, true);
+            }
+
+            /* switch to state TX */
+            dev->state = AT86RF215_STATE_TX;
+            dev->flags &= ~AT86RF215_OPT_TX_PENDING;
+            at86rf215_rf_cmd(dev, CMD_RF_TX);
+
+            /* This also tells the upper layer about retransmissions - should it be like that? */
```

Comment:
No idea about this one...

Reply:
@miri64 maybe you have an opinion here `NETDEV_EVENT_TX_STARTED` be notified on re-transmissions as well?